src/UI/ValueFactory.cs

```diff
@@ -388,5 +406,33 @@ private static bool GetNewColorSchemeValue(Design design, out object? newValue)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Returns true if path is to a directory.
+        /// Thanks: https://stackoverflow.com/a/19596821/4824531
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        static bool IsPathDirectory(string path)
+        {
+            if (path == null) throw new ArgumentNullException("path");
+            path = path.Trim();
+
+            if (Directory.Exists(path))
+                return true;
+
+            if (File.Exists(path))
+                return false;
+
+            // neither file nor directory exists. guess intention
+
+            // if has trailing slash then it's a directory
+            if (new[] { Path.DirectorySeparatorChar,Path.AltDirectorySeparatorChar}.Any(x => path.EndsWith(x)))
```

Comment:
This is invalid, depending on platform and file system. For example, a backslash character is a perfectly valid path component on most linux filesystems.

Here's on a standard ext4 partition on Ubuntu 23.10:
```
:~/dev/test$ mkdir This\ Ends\ In\ A\ Backslash\\
:~/dev/test$ ls -1
Program.cs
test.csproj
'This Ends In A Backslash\'
```

Because backslash is a legal name component in unixy systems, Directory.Exists should be trusted on them, and we shouldn't second-guess the user. If they put in a backslash on \*nix, that's on them.

If you reeeeaaaaaalllllly want to do this, anyway, do a platform check first or make this method platform-conditional (.net 8 has sweet new ways to do that). I'd just suggest not doing it at all though. Windows is capable of dealing with those paths, too, if, say, it mounts an NFS share from a linux box that has a path like that or if the `\\?\` extended path syntax is used (for certain other situations).

I think there are way too many edge cases to consider for this to be worth trying to do.